### PR TITLE
Backport faster T.35 payload copy from dav1d 1.2.1

### PR DESCRIPTION
**Pull request by fbossen**

### CODE REVIEW

File: src/getbits.rs [hunk 1]
```diff
@@ -197,4 +197,11 @@ impl<'a> GetBits<'a> {
     pub const fn has_pending_bits(&self) -> bool {
         self.state != 0 || self.bits_left != 0
     }
+
+    pub fn get_bytes(&mut self, n: usize) -> Box<[u8]> {
```
Comment (kkysen) [inline src/getbits.rs:201]:
```suggestion
    pub fn get_bytes(&mut self, n: usize) -> &[u8] {
```

File: src/getbits.rs [hunk 2]
```diff
@@ -197,4 +197,11 @@ impl<'a> GetBits<'a> {
...
+        assert_eq!(self.bits_left, 0);
+        let i = self.index;
+        self.index += n;
+        self.data[i..][..n].into() // TODO(kkysen) fallible allocation
```
Comment (kkysen) [inline src/getbits.rs:205]:
```suggestion
        &self.data[i..][..n].into()
```

File: src/obu.rs [hunk 3]
```diff
@@ -2462,7 +2462,7 @@ unsafe fn parse_obus(
                     } else {
                         let country_code = country_code as u8;
                         let country_code_extension_byte = country_code_extension_byte as u8;
-                        let payload = (0..payload_size).map(|_| gb.get_bits(8) as u8).collect(); // TODO(kkysen) fallible allocation
+                        let payload = gb.get_bytes(payload_size as usize);
```
Comment (kkysen) [inline src/obu.rs:2465]:
```suggestion
                        // TODO fallible allocation
                        let payload = gb.get_bytes(payload_size as usize).into();
```

File: src/getbits.rs (hunk 1 above)
Comment (kkysen) [reply src/getbits.rs:201]: I think it's better to leave it up to the caller if they want to allocate or not.

**Review by kkysen — CHANGES_REQUESTED** [review]
I'd change `fn get_bytes` to not allocate and allocate in the caller instead, but otherwise LGTM.

File: src/getbits.rs (hunk 1 above)
Comment (fbossen) [reply src/getbits.rs]: Agreed

**Review by kkysen — APPROVED** [review]

**Pull request merged**